src/GameEngine/GameEngineDriver.cpp

```diff
@@ -149,5 +149,40 @@ namespace GameEngine {
             return true;
         }
 
+        bool test_tournament(){
+            Terminal::test_mode_on();
+            auto tournament_1 = new Tournament();
```

Comment:
I think that's a memory leak since there's no delete at the end

Reply:
Well at the end there is 
`GameEngine::instance()->game_state->clear();`
`GameEngine::instance()->reset_test();`

I found that if I add `delete tournament_1;` before those two lines, the program executes till the end but then crashes. However, If I write the delete command, keep the game_state->clear() and ommit the rest_test then the program executes and exits as intended.